omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/PatientAllergyController2_0Test.java

```diff
@@ -244,6 +247,18 @@ public void shouldThrowExceptionWhenSettingNoKnownAllergiesIfAllergiesNotEmpty()
 		// attempt to set no known allergies with PUT
 		handle(newPutRequest(getURI(), json));
 	}
+
+	/**
+	 * Return no response body with a 204 http response status code for a patient with UNKNOWN Allergies
+	 */
+	@Test(expected = NoContentFoundException.class)
+	public void shouldEnsureFetchingAllergiesForPatientReturnsNoContentResponseStatusCode() throws Exception {
+		handle(newDeleteRequest(getURI(), new Parameter("reason", "unit test")));
+		
+		MockHttpServletResponse response = handle(newGetRequest(getURI()));
+		
+		Assert.assertEquals(HttpStatus.NO_CONTENT.value(), response.getStatus());
```

Comment:
If the`NoContentFoundException` is thrown, will the above line ever be reached?

Reply:
i believe when the [NoContentFoundException](https://github.com/openmrs/openmrs-module-webservices.rest/blob/d1ac1e1f868249f87cc613ef3220bb63882682f9/omod-common/src/main/java/org/openmrs/module/webservices/rest/web/response/NoContentFoundException.java) is thrown [here](https://github.com/openmrs/openmrs-module-webservices.rest/blob/d1ac1e1f868249f87cc613ef3220bb63882682f9/omod-2.0/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs2_0/PatientAllergyResource2_0.java#L172), its an indication that the response status code is a [204](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/204) resulting from the patient lacking an allergy.

Reply:
Did you try run the test in debug mode and confirmed that line 260 is hit?

Reply:
lemi try that, thanks!

Reply:
> If the`NoContentFoundException` is thrown, will the above line ever be reached?

when i remove `@Test(expected = NoContentFoundException.class)` from the test, the test fails when `NoContentFoundException` is thrown.

Reply:
You still have not answered the question.

Reply:
to answer the question, the mentioned line (260) wouldnt be reached. on going through the reporters talk post, i need to return a `204 No Content with empty body` which i have only achieved halfway. does that answer the question @dkayiwa ?

Reply:
Is the `Assert.assertEquals` on line 260 ever executed?

Reply:
yes it is

Reply:
Can you run the test in debug mode and confirm that you can step over this line?

Reply:
i ran the test in debug mode as earlier suggested, i added a toggle breakingpoint but nothing happens. everything is blank. no failure trace given nothing. 

this goes to all other tests tht i tried to do the same.

Reply:
The other alternative would be to temporarily put `System.out.println()` on various lines and see the out put on the console to see which lines were run and which were not. With that, you do not have to run in debug mode.

Reply:
i have tried this approach but for some reason i dont get any results in the console. I could be doing something wrong

Reply:
If you removed the throwing of the exception on line 172 of PatientAllergyResource2_0.java, are you then able to get the results on the console?

Reply:
when i remove the throwing on line 172, i get this error in the console which says `Caused by: java.lang.AssertionError: expected:<204> but was:<200>`.






The whole failure trace is shown below:
```
java.lang.Exception: Unexpected exception, expected<org.openmrs.module.webservices.rest.web.response.NoContentFoundException> but was<java.lang.AssertionError>
	at org.junit.internal.runners.statements.ExpectException.evaluate(ExpectException.java:28)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.springframework.test.context.junit4.statements.RunBeforeTestMethodCallbacks.evaluate(RunBeforeTestMethodCallbacks.java:73)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.springframework.test.context.junit4.statements.RunAfterTestMethodCallbacks.evaluate(RunAfterTestMethodCallbacks.java:82)
	at org.springframework.test.context.junit4.statements.SpringRepeat.evaluate(SpringRepeat.java:73)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271)
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.runChild(SpringJUnit4ClassRunner.java:217)
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.runChild(SpringJUnit4ClassRunner.java:83)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229)
	at org.springframework.test.context.junit4.statements.RunBeforeTestClassCallbacks.evaluate(RunBeforeTestClassCallbacks.java:61)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.springframework.test.context.junit4.statements.RunAfterTestClassCallbacks.evaluate(RunAfterTestClassCallbacks.java:68)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:309)
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.run(SpringJUnit4ClassRunner.java:163)
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:93)
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:40)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:529)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:756)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:452)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:210)
Caused by: java.lang.AssertionError: expected:<204> but was:<200>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:743)
	at org.junit.Assert.assertEquals(Assert.java:118)
	at org.junit.Assert.assertEquals(Assert.java:555)
	at org.junit.Assert.assertEquals(Assert.java:542)
	at org.openmrs.module.webservices.rest.web.v1_0.controller.openmrs2_0.PatientAllergyController2_0Test.shouldEnsureFetchingAllergiesForPatientReturnsNoContentResponseStatusCode(PatientAllergyController2_0Test.java:260)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.ExpectException.evaluate(ExpectException.java:19)
	... 24 more


```

Reply:
Doesn't that confirm that the line is not executed with your changes to fix the reported problem on the ticket?

Reply:
Test method expects a `204` response which is provided for in the `nocontentfoundexception()` method which results into the test being green. However when i remove the exception from [line 172](https://github.com/openmrs/openmrs-module-webservices.rest/blob/d1ac1e1f868249f87cc613ef3220bb63882682f9/omod-2.0/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs2_0/PatientAllergyResource2_0.java#L172), i get a `200` reponse which results into the test failing. My thinking is, the line 260 in the test method is hit

Reply:
> My thinking is, the line 260 in the test method is hit

Basing on how your test is written, line 260 is not hit unless you can prove so.

Reply:
> test is written

look through again please